server/slack_notifier_test.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/applikatoni/applikatoni/models"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestNotifySlack(t *testing.T) {
+	target := &models.Target{Name: "staging"}
+
+	application := &models.Application{
+		GitHubOwner: "shipping-co",
+		GitHubRepo:  "main-web-app",
+	}
+
+	deployment := &models.Deployment{
+		ApplicationName: "web",
+		TargetName:      target.Name,
+		Branch:          "master",
+		CommitSha:       "f00b4r",
+	}
+
+	expectedMessage := slackMsg{
+		Username:  "Applikatoni Bot",
+		IconEmoji: ":it:",
+		Text: fmt.Sprintf("Successfully deployed %s :tada:\n to: %s \n on: %s %s",
+			deployment.ApplicationName,
+			deployment.TargetName,
+			deployment.Branch,
+			deployment.CommitSha),
+	}
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		body, _ := ioutil.ReadAll(r.Body)
+		expectedJson, _ := json.Marshal(expectedMessage)
+		if string(body) != string(expectedJson) {
+			t.Errorf("sent wrong payload expected=%v got=%v", expectedMessage, string(body))
```

Comment:
ah, yes, you are right. the successfully notified was what confused me. I'll have it return a 422 for Unprocessable Entity.